### DISCUSSION TITLE
Only one IP from two will be used

### DISCUSSION THREAD

**Discussion by Pacerino** [postalserver/postal#2728] [Help with using Postal] [answered]
We now use Postal to send mails from our system. Approximately 60,000 every day. We have two IP addresses and have therefore configured the IP pool. The problem is that Yahoo has now blocked our second IP. On closer inspection, I noticed that we only ever send to Yahoo with this one IP, although the configuration of the IP pools is 50 / 75%. 

Google Postmaster also only shows the second IP in the IP reputation. Strangely enough, this IP is the second IP that we have subsequently configured. Before that, the server was already running perfectly with the first one. What can I do to solve the problem?

- OS: Debian GNU/Linux 12 (bookworm)
- Postal Version: 2.1.2.

Edit:
I've seen that in the docs its described that the IP needs to be assigned to the Server. My Postal Installation is running on a virtual host and got its IP from the Router. Is it possible to route the IP from OpnSense to the Postal installation with like VirtualIP, VLAN or something else?

**Comment by willpower232** [comment]
Postal needs to be able to see the IP addresses you would like it to use, if there is some kind of internal/external mapping then that should be fine too as long as you configure Postal to use the internal IP addresses it can see.

If you use `ip a` on the virtual host running Postal, you should be able to see the available IP addresses. If one is missing then that might explain your issue. If you can see both present there and the IP Pool definitely contains both then it might be worth playing around with the configuration of the IP Pool `50 / 75%`

**Reply by Pacerino** [reply]
So after some testing with two NIC and following setup i cant get it to work.

- ens18 is the LAN IP 192.168.10.2 and NAT(ted) via OpnSense to its public IP
- ens19 is directly mapped to the VM, so no LAN, directly to the public IP 2

In Postal I've added the LAN address and the second public IP. Mails are going out, but the IP stays always the same. ens18 will be used. I've then tried to remove the ens18 address completely from the IP Pool and only assign the ens19 address. Still the same, the ens18 address will be used. Postal can see both addresses, both addresses can be used. Testing with curl and define the interfaces there directly shows that on the host site everything works.

I would appreciate any help!

```
curl --interface ens18 ifconfig.me
116.202.XXX.XXX

curl --interface ens19 ifconfig.me
162.55.XXX.XXX
```
Attached is a quick and dirty drawing of our setup.

![Setup](https://github.com/postalserver/postal/assets/22194063/dc540709-3d66-499c-a98d-524b2d335312)

Please don' ask why this setup is so complicated... If i could make decisions i wouldn't put the Postal VM behind a Router.

**Reply by Pacerino** [reply, marked as answer]
I don't know a bit of Ruby, but you get the idea. I have now spent 2 hours trying to understand the code and came across `ip_pool_for_message`. Since I haven't created any rules for the IP pool yet, the IP pool was simply ignored. By creating a rule, I could now successfully send via ens19. I then looked in the documentation to see if this is documented anywhere. Result: No!
